extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java

```diff
@@ -127,11 +125,12 @@
  *
  * @author aaron
  */
-@Path ( "/repo/files/" )
+@Path( "/repo/files/" )
 public class FileResource extends AbstractJaxRSResource {
 
   private static final String INVALID_SECURITY_PRINCIPAL_CHARACTERS = ".*[#,+\"\\\\<>]+.*";
-  private static final Pattern INVALID_SECURITY_PRINCIPAL_PATTERN = Pattern.compile( INVALID_SECURITY_PRINCIPAL_CHARACTERS );
+  private static final Pattern INVALID_SECURITY_PRINCIPAL_PATTERN =
```

Comment:
Can we track this with a jira task or if we are sure add a `@java.lang.SuppressWarnings`
![regex_possible_dos](https://github.com/pentaho/pentaho-platform/assets/2119761/13d6efcc-f61b-4494-82e9-f9a5b922a7ed)


this came up on the sonarqube only because you change the formatting

 "Make sure the regex used here, which is vulnerbale to polynomial runtime due to backtracking, cannot lead to denail of service"

Reply:
Removed the formatting changes from this file.
Will create a JIRA issue to address the DOS discovered (which I confirmed is real).

Reply:
https://hv-eng.atlassian.net/browse/PPP-5105